src/main/java/org/apache/brooklyn/cloudfoundry/entity/VanillaCloudfoundryApplicationImpl.java

```diff
@@ -235,10 +354,60 @@ public void restart() {
     public void destroy() {
         super.destroy();
         disconnectSensors();
+        getCloudFoundryLocation().delete(applicationName);
     }
 
     protected CloudFoundryPaasLocation getCloudFoundryLocation() {
         return cfLocation;
     }
 
+
+    private void waitForEntityStart() {
+        if (log.isDebugEnabled()) {
+            log.debug("waiting to ensure {} doesn't abort prematurely", this);
+        }
+        Duration startTimeout = getConfig(START_TIMEOUT);
```

Comment:
instead of using a CoundownTimer, does the `cf-client` offers a `getApplicationStatus` ?